index.js

```diff
@@ -0,0 +1,63 @@
+'use strict';
+require('env2')('.env');
+
+const Promise = require('bluebird');
+const sauce = require('mintsauce');
```

Comment:
So... the idea of middleware chains for lambda came out of conversations with Pascal about how we could re-use more code across providers, since all were performing similar tasks.

More generally, the advantages of implementing a middleware chain are:

* flattening the control flow of asynchronous functions makes for easier code. To do the same with "plain" callbacks would involve some boilerplate code, which is effectively what has been abstracted into the `mintsauce` library. 
* "for free" error handling - note the absence of `if (err) return callback (err)` or similar code.
* common method signatures and arguments make the task of creating re-usable components easier since all middleware functions take the same arguments.
* "purity" of functions (in an f.p. sense) makes testing components easier, since no component is dependent on any scope beyond its own arguments.